junit5-jupiter-starter-bazel/junit5.bzl

```diff
@@ -0,0 +1,100 @@
+"""External dependencies & java_junit5_test rule"""
+
+JUNIT_JUPITER_GROUP_ID = "org.junit.jupiter"
+JUNIT_JUPITER_ARTIFACT_ID_LIST = [
+    "junit-jupiter-api",
+    "junit-jupiter-engine",
+    "junit-jupiter-params",
+]
+
+JUNIT_PLATFORM_GROUP_ID = "org.junit.platform"
+JUNIT_PLATFORM_ARTIFACT_ID_LIST = [
+    "junit-platform-commons",
+    "junit-platform-console",
+    "junit-platform-engine",
+    "junit-platform-launcher",
+    "junit-platform-suite-api",
+]
+
+JUNIT_EXTRA_DEPENDENCIES = [
+    ("org.apiguardian", "apiguardian-api", "1.0.0"),
+    ("org.opentest4j", "opentest4j", "1.1.0"),
+]
+
+
+def junit_jupiter_java_repositories(
+        version="5.2.0"):
+    """Imports dependencies for JUnit Jupiter"""
+    for artifact_id in JUNIT_JUPITER_ARTIFACT_ID_LIST:
+        native.maven_jar(
+            name=_format_maven_jar_name(JUNIT_JUPITER_GROUP_ID, artifact_id),
+            artifact="%s:%s:%s" % (
+                JUNIT_JUPITER_GROUP_ID, artifact_id, version)
+        )
+    
+    for t in JUNIT_EXTRA_DEPENDENCIES:
+        native.maven_jar(
+            name=_format_maven_jar_name(t[0], t[1]),
+            artifact="%s:%s:%s" % t
+        )
+
+
+def junit_platform_java_repositories(
+        version="1.2.0"):
+    """Imports dependencies for JUnit Platform"""
+    for artifact_id in JUNIT_PLATFORM_ARTIFACT_ID_LIST:
+        native.maven_jar(
+            name=_format_maven_jar_name(JUNIT_PLATFORM_GROUP_ID, artifact_id),
+            artifact="%s:%s:%s" % (
+                JUNIT_PLATFORM_GROUP_ID, artifact_id, version)
+        )
+
+
+def java_junit5_test(name, srcs, test_class=None, deps=[], runtime_deps=[], **kwargs):
+    FILTER_KWARGS = [
+        "main_class",
+        "use_testrunner",
+        "args",
+    ]
+
+    for arg in FILTER_KWARGS:
+        if arg in kwargs.keys():
+            kwargs.pop(arg)
+
+    junit_console_args = []
+    if test_class:
+        junit_console_args += ["--select-class", test_class]
+    else:
+        fail("must specific 'test_class'")
+
+    native.java_test(
+        name=name,
+        srcs=srcs,
+        use_testrunner=False,
+        main_class="org.junit.platform.console.ConsoleLauncher",
+        args=junit_console_args,
+        deps=deps + [

```

Comment:
I could be wrong about this, but I think the lack of spaces around these `=` signs aren't idiomatic.

Have you considered (or already done) running buildifier from https://github.com/bazelbuild/buildtools to format the BUILD files uniformly?

Reply:
That's cool, I just formatted it with autopep8.